compendium/DeclarativeServices/src/manager/BindingPolicyDynamicReluctant.cpp

```diff
@@ -0,0 +1,78 @@
+/*=============================================================================
+
+  Library: CppMicroServices
+
+  Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+  file at the top-level directory of this distribution and at
+  https://github.com/CppMicroServices/CppMicroServices/COPYRIGHT .
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  =============================================================================*/
+
+#include "ReferenceManagerImpl.hpp"
+
+namespace cppmicroservices {
+namespace scrimpl {
+
+using namespace cppmicroservices::logservice;
+
+void ReferenceManagerBaseImpl::BindingPolicyDynamicReluctant::ServiceAdded(
+  const ServiceReferenceBase& reference)
+{
+  std::vector<RefChangeNotification> notifications;
+  if (!reference) {
+    Log("BindingPolicyDynamicReluctant::ServiceAdded called with an invalid "
+        "service reference");
+    return;
+  }
+
+  auto notifySatisfied = false;
+
+  if (!mgr.IsSatisfied()) {
+    notifySatisfied =
+      mgr.UpdateBoundRefs(); // becomes satisfied if return value is true
+  } else {
+    // previously satisfied
+    // If the service was previously satisfied then either there is
+    // nothing to do or a rebind needs to happen if the cardinality
+    // is optional and there are no bound refs.
+    if (mgr.IsOptional() && 0 == mgr.GetBoundReferences().size()) {
```

Comment:
calling `IsOptional` doesn't appear to be necessary. If `GetBoundReferences` returns 0 then that implies that the service reference is optional (since we're in the `else` because the service reference was previously satisfied).